### DISCUSSION TITLE
Segmentation find a part of big object

### DISCUSSION THREAD

**Discussion by ffamoss** [ultralytics/ultralytics#3445] [Q&A]
Hi. I try to  train custom segmentation model. Play with different parameters like:
`yolo task=segment mode=train model=yolov8s-seg.pt data=polygon_docker.yaml device=0,1 epochs=100 imgsz=640 single_cls=True cache=True degrees=0.0 scale=0.0 perspective=0.0 max_det=1000`

But i can't fix issue like this
![pic1](https://github.com/ultralytics/ultralytics/assets/1873712/79348da1-eab1-4c99-89bf-f78c52cf755d)
![pic2](https://github.com/ultralytics/ultralytics/assets/1873712/1586267e-bd64-4f48-a904-c4e8d8e70e87)

What parameters i should change to fix this issue?

**Comment by glenn-jocher** [comment]
@ffamoss hello! It looks like you're working on training a custom segmentation model with YOLOv8 and encountering some issues with segmenting parts of a larger object. Segmentation tasks can be quite sensitive to various hyperparameters and training data quality.

Here are a few suggestions that might help improve your segmentation results:

1. **Data Quality**: Ensure that your dataset annotations are accurate and consistent. High-quality, precise annotations are crucial for segmentation tasks.

2. **Augmentations**: You might want to experiment with different data augmentation strategies. While you've turned off some augmentations (degrees, scale, perspective), sometimes introducing controlled augmentations can help the model generalize better to different conditions.

3. **Model Complexity**: If your model is too simple, it might not capture the complexity of the task. Consider using a larger model if your computational resources allow it.

4. **Learning Rate**: Adjusting the learning rate can sometimes lead to better convergence. A learning rate scheduler that adapts the rate over training epochs might be beneficial.

5. **Loss Function**: Review the loss function being used for segmentation. Some loss functions might be more suitable for your specific task, especially if you're dealing with class imbalances or particular characteristics in your data.

6. **Post-Processing**: Sometimes, post-processing steps like Conditional Random Fields (CRFs) can refine the segmentation results, especially around the edges of objects.

7. **Batch Size**: A larger batch size can provide more stable gradient estimates during training, but it's limited by your GPU memory. If possible, try increasing the batch size.

8. **Epochs**: More training epochs might be necessary for the model to learn to segment the object parts correctly, especially if the dataset is complex.

Remember, it's often a process of trial and error to find the right combination of parameters and settings. Keep experimenting, and don't hesitate to check the [Segmentation Docs](https://docs.ultralytics.com/tasks/segment) for more detailed guidance. Good luck with your model training! 🚀
